TrafficCapture/trafficReplayer/src/main/java/org/opensearch/migrations/replay/datatypes/UniqueRequestKey.java

```diff
@@ -1,16 +1,34 @@
 package org.opensearch.migrations.replay.datatypes;
 
 public class UniqueRequestKey {
-    public final ITrafficStreamKey trafficStreamKey;
-    public final int requestIndex;
+    public final TrafficStreamKeyWithRequestOffset trafficStreamKeyAndOffset;
+    public final int replayerRequestIndex;
 
-    public UniqueRequestKey(ITrafficStreamKey streamKey, int requestIndex) {
-        this.trafficStreamKey = streamKey;
-        this.requestIndex = requestIndex;
+    public UniqueRequestKey(ITrafficStreamKey streamKey, int sourceOffset, int replayerIndex) {
+        this(new TrafficStreamKeyWithRequestOffset(streamKey, sourceOffset), replayerIndex);
+    }
+
+    public UniqueRequestKey(TrafficStreamKeyWithRequestOffset keyWithRequestOffset, int replayerIndex) {
+        this.trafficStreamKeyAndOffset = keyWithRequestOffset;
+        this.replayerRequestIndex = replayerIndex;
+    }
+
+    public int getSourceRequestIndex() {
```

Comment:
When will we use source request index vs replayer request index?

Reply:
Both are used within logs for identification and tracing.  ReplayerRequestIndex is used to resequence in RequestSenderOrchestrator
```
                                                        cffr.scheduleSequencer.add(requestKey.getReplayerRequestIndex(),
                                                                () -> successFn.accept(channelFutureAndRequestSchedule),
                                                                x -> x.run());
```